clconf/template_test.go

```diff
@@ -0,0 +1,375 @@
+		"yes_basedir.html.clconf",
+		"yes_basedir1.html.clconf",
+		"yes_basedir2.html.clconf",
+	})
+}
+
+func TestFindTemplatesSingleFileMatchingExtension(t *testing.T) {
+	testFindTemplates(t, "Single File Matching Extension", ".clconf", "subdir1/subsubdir1/yes_subdir1subsubdir1.sh.clconf", true, []string{
+		"subdir1/subsubdir1/yes_subdir1subsubdir1.sh.clconf",
+	})
+}
+
+func TestFindTemplatesSingleFileNotMatchingExtension(t *testing.T) {
+	testFindTemplates(t, "Single File Not Matching Extension", ".clconf", "subdir1/subsubdir1/no_subdir1subsubdir1.sh", true, []string{
+		"subdir1/subsubdir1/no_subdir1subsubdir1.sh",
+	})
+}
+
+func TestFindTemplatesSingleEmptyFolder(t *testing.T) {
+	testFindTemplates(t, "Empty Folder", ".clconf", "emptydir", true, []string{})
+}
+
+func defaultContext() (*TemplateSettings, *SecretAgent) {
+	return &TemplateSettings{
+		Flatten:           false,
+		KeepExistingPerms: false,
+		Rm:                false,
+		DirMode:           "753", //We use a wonky value to look for it later
+		Extension:         ".clconf",
+	}, NewSecretAgent([]byte(""))
+}
+
+func exists(path string) bool {
+	_, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func checkFile(t *testing.T, path string, expectedPerms os.FileMode) {
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		t.Fatalf("Error reading %q: %v", path, err)
+	}
+
+	stat, _ := os.Stat(path)
+	if stat.Mode() != expectedPerms {
+		t.Fatalf("File %q has mode %o that does not match expected value %o", path, stat.Mode(), expectedPerms)
+	}
+
+	if string(content) != fakeValue {
+		t.Fatalf("File %q content %q does not match expected %q", path, content, fakeValue)
+	}
+}
+
+func TestProcessTemplateInPlace(t *testing.T) {
+	extension := ".clconf"
+	temp := buildTestFolder(t)
+	defer os.RemoveAll(temp)
+
+	context, secretAgent := defaultContext()
+
+	value := map[interface{}]interface{}{"foo": fakeValue}
+
+	template := filepath.Join(temp, "yes_basedir.html"+extension)
+	_, err := context.processTemplate(pathWithRelative{
+		fullPath: template,
+		relPath:  filepath.Base(template),
+	}, "", value, secretAgent)
+	if err != nil {
+		t.Fatalf("processTemplate reported error: %v", err)
+	}
+
+	checkFile(t, filepath.Join(temp, "yes_basedir.html"), 0646)
+
+	if !exists(template) {
+		t.Errorf("Template went missing when it wasn't supposed to!")
+	}
+
+	context.FileMode = "610"
+	_, err = context.processTemplate(pathWithRelative{
+		fullPath: template,
+		relPath:  filepath.Base(template),
+	}, "", value, secretAgent)
+	if err != nil {
+		t.Fatalf("processTemplate reported error: %v", err)
+	}
+
+	checkFile(t, filepath.Join(temp, "yes_basedir.html"), 0610)
+
+	context.Rm = true
+	context.FileMode = "601"
+	context.KeepExistingPerms = true
+	_, err = context.processTemplate(pathWithRelative{
+		fullPath: template,
+		relPath:  filepath.Base(template),
+	}, "", value, secretAgent)
+	if err != nil {
+		t.Fatalf("processTemplate reported error: %v", err)
```

Comment:
This is the exact same error message as the previous.  Any failures with this message will have no indication of which one failed.  At minimum the error messages need to differ.

This is likely a good place to use:
```go
func TestProcessTemplateInPlace(t *testing.T) {
    t.Run("without rm", func(t *testing.T) { ... })
    t.Run("with rm", func(t *testing.T) { ... })
}
```
or even make them independent test functions altogether.

_Multiple occurrences_

Reply:
[Fixed](https://github.com/pastdev/clconf/pull/11/commits/04b8136014f5cc7f0e438636230676687d97b17f)